src/Nethermind/Nethermind.Evm/EOF/EvmObjectFormat.cs

```diff
@@ -261,6 +262,52 @@ public bool ValidateBody(ReadOnlySpan<byte> container, in EofHeader header)
                 return false;
             }
 
+            for (int sectionIdx = 0; sectionIdx < header.CodeSections.Length; sectionIdx++)
+            {
+                SectionHeader sectionHeader = header.CodeSections[sectionIdx];
+                (int codeSectionStartOffset, int codeSectionSize) = (sectionHeader.Start, sectionHeader.Size);
+                ReadOnlySpan<byte> code = container.Slice(codeSectionStartOffset, codeSectionSize);
+                if (!ValidateInstructions(code, header))
+                {
+                    if (Logger.IsTrace) Logger.Trace($"EIP-3670 : CodeSection {sectionIdx} contains invalid body");
+                    return false;
+                }
+            }
+
+            return true;
+        }
+        bool ValidateInstructions(ReadOnlySpan<byte> code, in EofHeader header)
+        {
+            for (int pos = 0; pos < code.Length; pos++)
+            {
```

Comment:
Perf considerations - does it make sense to make any of this for loops a Parallel.For loop, considering the limited size of contract code? Maybe some benchmarks? Or some heuristic that would decide it based on the lengths?

PUSH instruction would make it hard on the instruction level though.

Reply:
Can you check this PR https://github.com/NethermindEth/nethermind/pull/5099 ? 
it has some benchmarks from benchmarkDotNet and it uses ParallelFor